### PR TITLE
Fix Json serialization for Tuples of enum values

### PR DESCRIPTION
**Pull request by ggleyzer**
Fix serialization of tuples containing enum values, such as:
    
    Tuple<Int, Boolean> tuple = (1, True)

Prior to this change json serialization would blow up at run time.

### CODE REVIEW

File: javatools/src/main/java/org/xvm/asm/constants/ParameterizedTypeConstant.java [hunk 1]
```diff
@@ -814,6 +808,34 @@ public ResolutionResult resolveContributedName(
         return result;
         }
 
+    @Override
+    public TypeConstant widenEnumValueTypes()
+        {
+        TypeConstant   constUnderlying = m_constType;
+        boolean        fDiff           = false;
```
Comment (cpurdy) [inline javatools/src/main/java/org/xvm/asm/constants/ParameterizedTypeConstant.java:815]: `fDiff` will always be the same as `(aconstOriginal != aconstResolved)`

**Review by cpurdy — COMMENTED** [review]
Reviewed.

**Review by cpurdy — APPROVED** [review]
Approved.

File: javatools/src/main/java/org/xvm/asm/constants/ParameterizedTypeConstant.java (hunk 1 above)
Comment (ggleyzer) [reply javatools/src/main/java/org/xvm/asm/constants/ParameterizedTypeConstant.java]: Changed accordingly

**Pull request merged**